homeassistant/components/light/xiaomi_miio.py

```diff
@@ -168,6 +171,74 @@ def async_turn_on(self, **kwargs):
             if result:
                 self._brightness = brightness
 
+        result = yield from self._try_command(
+            "Turning the light on failed.", self._light.on)
+
+        if result:
+            self._state = True
```

Comment:
As try_command already returns true on success and false on failure, the above result from a yield could be directly `self._state` maybe? Applies also other parts of the code.

Reply:
Good proposal. I will care about!